### ISSUE TITLE
What to discuss with TA

### ISSUE THREAD

**Issue by juimdpp** [swsnu/swpp2020-team3#1]
Add questions to ask

> Questions that have been handled
> **answers, replies, comments**

**Comment by juimdpp** [comment]
> 새로고침 핸들링할때 서버에서 항상 다 불러오는지, cookie, local storage를 불러오는지... because store 쓰면 안 됨
> 
> **==>서버에서 불러온다. 대신 쿼리를 잘 날려야함.**

**Comment by juimdpp** [comment]
> 이미지 삽입할 때, 크기 어떻게 맞출지... (사진마다 크기가 다를텐데 하나로 키우거나 줄이면 회상도가 안 맞아서 보기 안 좋을 수 있을텐데, 이건 어떻게 해결할지..?  사이즈 안 맞으면 검은/흰 배경 위에 맞춰야하나)

> **구현에 따라. 너무 크면 적게 만들면 됨. 너무 작으면 키우면 됨.**

**Comment by juimdpp** [comment]
> - model ER-diagram 확인해주기 **done**
> - child comment should be in a different model? **That is what we did. She didn't say anything bad about the implementation, so good.**
> - user extend best method?? **there exists, using anonymous user. the below link is good**
> https://simpleisbetterthancomplex.com/tutorial/2016/07/22/how-to-extend-django-user-model.html

**Comment by juimdpp** [comment]
> - image field in model 테이블
> - local storage or online url when storing images
> 
> **must use server to communicate with the users, so must save them locally**

**Comment by juimdpp** [comment]
How to control versions..?

**Comment by juimdpp** [comment]
> Description_step 모델을 따로 만드는 것이 나을 것인지, 아니면 지금처럼 recipe model 안에서 description_list, image_list를 가지고 가는 것이 좋을지.
> Not very intuitive, array becomes one entire textfield.
> 
> **decided to use JSONField()**

**Comment by juimdpp** [comment]
> How can we reinforce one-to-many relationship for arraylist?
> For example, in our case, each article can have multiple comments.
> ... finish this question.. ㅎ
>
> ==> **Use foreignkey to reinforce array list**

**Comment by juimdpp** [comment]
> 이미지 파일 업로드: 현재 보다 더 좋은 방법이 있는지.
> 현재: send as DataURL and decode using base64. Apparently this method is slow (documentation?)
> suggestion?:  I read that Django provides request.FILES.getlist() in order to get files. However, all the examples i've seen online use forms. What exactly are forms? How do they differ from react components? Study and ask for the best.
> **No issues with security. Choose the best for us. But, make sure you consider performance**

**Comment by juimdpp** [comment]
> What is the best method to handle login? 
> Retrieve from server list of logged in users.
> Is it the localStorage? 
> Cookies? How do we use cookies? Is it using token?
> **TLDR: use server (first method)**
> **Localstorage isn't secure**
> **Cookie is deprecated**

**Comment by juimdpp** [comment]
Is it ok to add a new feature (about meal planning)?

**Comment by juimdpp** [comment]
> 발표 준비한것 보여드리기:
> - 구현한것 (데모) --> 어떤것을 개선 시켜야할까
> - 발표에서 자주 나오는 질문
> - 발표 자료에 더 들어갔으면 하는 내용
> **질문: machine learning api, how we will crawl recipes and ingredients...**
